crates/bin/cli-wallet/src/sync.rs

```diff
@@ -0,0 +1,208 @@
+        };
+
+        let node_client = wallet::connect_to_node(&node_url)
+            .await
+            .map_err(|e| anyhow!("Failed to connect to node {}: {}", node_url, e))?;
+
+        Ok(Self {
+            node_client,
+        })
+    }
+}
+
+pub async fn sync_all_pending_operations(
+    pool: Pool<SqliteConnectionManager>,
+    _poll_interval: u64,
+    _timeout: u64,
+) -> Result<()> {
+    let (nodes, pending_mint_quotes, pending_melt_quotes) = {
+        let db_conn = pool.get()?;
+        let nodes = wallet::db::node::fetch_all(&db_conn)?;
+        let mint_quotes = wallet::db::get_pending_mint_quotes(&db_conn)?;
+        let melt_quotes = wallet::db::get_pending_melt_quotes(&db_conn)?;
+        (nodes, mint_quotes, melt_quotes)
+    };
+    
+    if nodes.is_empty() {
+        println!("No nodes registered");
+        return Ok(());
+    }
+
+    println!("Starting sync for {} nodes", nodes.len());
+
+    // Process nodes sequentially due to rusqlite::Connection not being Send + Sync
+    // TODO: Implement concurrent processing when wallet supports it
+    for (node_id, node_url) in nodes {
+        println!("Syncing node {} ({})", node_id, node_url);
+        
+        let mut node_sync = match NodeSync::new(pool.clone(), node_id).await {
+            Ok(sync) => sync,
+            Err(e) => {
+                eprintln!("Error connecting to node {}: {}", node_id, e);
+                continue;
+            }
+        };
+        
+        // Find quotes for this specific node to avoid processing unnecessary data
+        let node_mint_quotes = pending_mint_quotes
+            .iter()
+            .find(|(id, _)| *id == node_id)
+            .map(|(_, quotes)| quotes.as_slice())
+            .unwrap_or(&[]);
+            
+        let node_melt_quotes = pending_melt_quotes
+            .iter()
+            .find(|(id, _)| *id == node_id)
+            .map(|(_, quotes)| quotes.as_slice())
+            .unwrap_or(&[]);
+
+        // Sync mint quotes
+        if !node_mint_quotes.is_empty() {
+            if let Err(e) = sync_mint_quotes(&pool, &mut node_sync.node_client, node_id, node_mint_quotes).await {
+                eprintln!("Error syncing mint quotes for node {}: {}", node_id, e);
+            }
+        }
+        
+        // Sync melt quotes
+        if !node_melt_quotes.is_empty() {
+            if let Err(e) = sync_melt_quotes(&pool, &mut node_sync.node_client, node_id, node_melt_quotes).await {
+                eprintln!("Error syncing melt quotes for node {}: {}", node_id, e);
+            }
+        }
+        
+        println!("Completed sync for node {}", node_id);
+    }
+
+    println!("Sync completed for all nodes");
+    Ok(())
+}
+
+async fn sync_mint_quotes(
+    pool: &Pool<SqliteConnectionManager>,
+    node_client: &mut NodeClient<Channel>,
+    node_id: u32,
+    quotes: &[(String, String, i32, String, u64)],
+) -> Result<()> {
+    for (method, quote_id, previous_state, unit, amount) in quotes {
```

Comment:
remove the loop. Move it outside of the function. There is not upside of it being inside.

Reply:
just iterate over the pending mint_quote, and for each of them, call the `sync_mint_quote` function. Do it in `sync_all_pending_operations`